configs/config.go

```diff
@@ -16,14 +18,41 @@ type API struct {
 }
 
 type Terraform struct {
-	Bin    *string
-	Source *string
+	Bin            *string
+	Source         *string
+	TFInitOptions  *string `yaml:"tfInitOptions,omitempty"`
+	TFApplyOptions *string `yaml:"tfApplyOptions,omitempty"`
 }
 
 type DeployConfig struct {
 	UseCached       bool
 	DeployLocalPath string
 	BatchMode       bool
+	TFInitOptions   string
+	TFApplyOptions  string
 }
 
 var Regions = []string{"us-east-1", "us-east-2", "us-west-1", "us-west-2"}
+
+// Coalesce returns the first non-empty vluae, but takes into account a loading order,
+// which is CLI args > config file > environment variables > some default
+func Coalesce(arg *string, config *string, envvar *string, def *string) *string {
+
+	if arg != nil && len(*arg) > 0 {
```

Comment:
I wonder if you really want these `len(*arg) > 0` checks.
What if, for example, the dce.yaml file contains some tfApplyOptions, and you want to override those options with _no_ options?

As in, what happens if I run something like:

```
dce system deploy --tf-init-options ""
```

Reply:
I would handle that case there. This particular method though is designed to get the first non-null and non-empty string variable.